README.md

```diff
@@ -0,0 +1,153 @@
+# Semantic Cache
+
+Semantic Cache is a tool for caching natural text based on semantic similarity. It's ideal for any task that involves querying or retrieving information based on meaning, such as natural language classification or caching AI responses. Two pieces of text can be similar but not identical (e.g., "great places to check out in Spain" vs. "best places to visit in Spain"). Traditional caching doesn't recognize this semantic similarity and misses opportunities for reuse.
+
+Semantic Cache allows you to:
+
+- Easily classify natural text into predefined categories
+- Avoid redundant LLM work by caching AI responses
+- Reduce API latency by responding to similar queries with already cached values
+
+<img src="./assets/how-semantic-cache-works.png" width="700">
+
+## Highlights
+
+- **Uses semantic similarity**: Stores cache entries by their meaning, not just the literal characters
+- **Handles synonyms**: Recognizes and handles synonyms
+- **Complex query support**: Understands long and nested user queries
+- **Customizable**: Set a custom proximity threshold to filter out less relevant results
+
+## Getting Started
+
+### Prerequisites
+
+- An Upstash Vector database (create one [here](https://console.upstash.com/vector))
+
+### Installation
+
+After creating a vector database, you should install the repository using the following command.
+
+```bash
+pip install scaching
+```
+
+To use it in your project, you must import it at the beginning of your file:
+
+```python
+from upstash-semantic-cache import SemanticCache
```

Comment:
There is already a package that uses from semantic_cache import semantic_cache. Maybe it would be better if we change it to the upstash_semantic_cache. Is changing the folder name where semantic_caching.py is located enough for it? Then I can update this part to from upstash_semantic_cache import SemanticCache

Reply:
That could be, yes. It is a bit long, but I wouldn't mind that too much.

To do that, you need to change the folder name, and specify it in your pyproject.toml, like here https://github.com/upstash/qstash-py/blob/main/pyproject.toml#L30